packages/classic-webview/src/hooks/useModal.tsx

```diff
@@ -0,0 +1,26 @@
+import { createContext, useContext, useState } from 'react';
+
+export type Modal = 'unlock-hardcore' | undefined;
+
+type ModalContext = {
+  modal: Modal;
+  setModal: React.Dispatch<React.SetStateAction<Modal | undefined>>;
+};
+
+const modalContext = createContext<ModalContext | null>(null);
+
+export const ModalContextProvider = (props: { children: React.ReactNode }) => {
+  const [modal, setModal] = useState<Modal>(undefined);
+
+  return (
+    <modalContext.Provider value={{ modal, setModal }}>{props.children}</modalContext.Provider>
+  );
```

Comment:
I love the idea of using React Context to handle modal rendering! I think this could be improved by:

1. Having all modals render through this context. Not just hardcore. Doing so would ensure that only one modal is visible at a given time.

2. Make the context provider render the desired modal instead of relying on the context consumer to do it.  It also offloads all modal concerns to the context provider. This would also ensure that multiple components don't accidentally render the same modal.

3. I would update the `useModal` API to be `showModal("my-modal")` and `closeModal()`. `setModal(undefined)` doesn't describe what setting it to `undefined` means.